### PR TITLE
Removes the native library from the RoboRIO Jar

### PR DESCRIPTION
**Pull request by ThadHouse**
Includes the native zip in the maven artifact as well.

### CODE REVIEW

File: publish.gradle [hunk 1]
```diff
@@ -47,7 +47,13 @@ publishing {
...
                 artifact nat.networktablesJavadoc
                 if (project.buildArm) {
-                    artifact project('arm:ntcore').jar
+                    def natArm = project('arm:ntcore')
+                    artifact natArm.jar
+                    // If the library is not embedded include it in the repo
+                    if (!project.hasProperty('compilerPrefix')) {
+                      artifact natArm.ntcoreZip
+                    }
+
```
Comment (PeterJohnson) [inline publish.gradle]: Added extra empty line?

**Review by PeterJohnson — CHANGES_REQUESTED** [review]

File: java/java.gradle [hunk 2]
```diff
@@ -31,9 +31,11 @@ jar {
...
         binaries {
             withType(SharedLibraryBinarySpec) { binary ->
-                from(file(binary.sharedLibraryFile)) {
-                    into getPlatformPath(binary)
-                }
+                // Only include the native file if not cross compiling to the roboRIO
+                if (!project.isArm || project.hasProperty('compilerPrefix'))
+                  from(file(binary.sharedLibraryFile)) {
+                      into getPlatformPath(binary)
+                  }
```
Comment (PeterJohnson) [inline java/java.gradle]: Indentation 2 spaces instead of 4?

File: publish.gradle (hunk 1 above)
Comment (PeterJohnson) [reply publish.gradle]: Indentation of artifact is 2 spaces instead of 4

Comment (ThadHouse) [reply publish.gradle]: Fixed

File: java/java.gradle (hunk 2 above)
Comment (ThadHouse) [reply java/java.gradle]: Fixed

**Review by PeterJohnson — APPROVED** [review]

**Review by 333fred — APPROVED** [review]

**Pull request merged**